### PR TITLE
parse order by expression

### PR DESCRIPTION
**Pull request by meijies**
https://github.com/ballista-compute/ballista/issues/95

### CODE REVIEW

File: jvm/sql/src/main/kotlin/SqlParser.kt [hunk 1]
```diff
@@ -138,7 +138,13 @@ class SqlParser(val tokens: TokenStream) : PrattParser {
         groupBy = parseExprList()
       }
 
-      return SqlSelect(projection, filterExpr, groupBy, table.id)
+      // parse optional ORDER BY clause
+      var orderBy: List<SqlExpr> = listOf()
+      if (tokens.consumeKeywords(listOf("ORDER", "BY"))) {
+        orderBy = parseExprList()
```
Comment (andygrove) [inline jvm/sql/src/main/kotlin/SqlParser.kt:144]: This is a good start, but order by expressions can also have optional `ASC` or `DESC` qualifiers. For example, `ORDER BY start_date DESC, salary ASC`.

Comment (meijies) [reply jvm/sql/src/main/kotlin/SqlParser.kt]: Optional `ASC` or `DESC` has supported now. You can review again

File: jvm/sql/src/main/kotlin/Expressions.kt [hunk 2]
```diff
@@ -52,11 +52,14 @@ data class SqlAlias(val expr: SqlExpr, val alias: SqlIdentifier) : SqlExpr
 
 data class SqlCast(val expr: SqlExpr, val dataType: SqlIdentifier) : SqlExpr
 
+data class SqlSort(val expr: SqlExpr, val asc: String) : SqlExpr
```
Comment (andygrove) [inline jvm/sql/src/main/kotlin/Expressions.kt:55]: I think `asc` should be a boolean since there can only be two values - ascending or descending.

**Comment by andygrove** [comment]
Thanks @meijies this looks great! If you could just change `asc` to a boolean, I'll go ahead and merge this.

**Comment by meijies** [comment]
@andygrove the change has been done, you can review again

**Pull request merged**